### ISSUE TITLE
The app doesn't properly on android 8.0 Oreo

### ISSUE THREAD

**Issue by mataniss** [ITVlab/TvAppRepo#71]
The current version of the app doesn't work properly on android 8.0 Oreo or higher since it doesn't have permission to install apps from unknown sources. 
It's a new permission that was added at android 8.0.
To fix this problem, you have to add the following tag to the manifest:

    <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" />
 
I've downloaded the source code and added this tag and managed to run correctly this app on an android tv 8.0 emulator using android studio.

I hope you upload a new release for this app with this fix, It would be wonderful!

**Comment by Fleker** [comment]
That's pretty interesting.

I wonder how necessary it will be in Oreo, since you'll be able to pin arbitrary things to the homescreen.

Definitely something that should be fixed.

**Comment by mataniss** [comment]
Thank you for replying. I hope you fix this issue soon. I also can't seem to find the app on the play store, so if you can look into this issue. I'm running Android TV Oreo.

**Comment by Wanderer189** [comment]
I dont think anyone is working on this anymore.

@mataniss I tried to disassemble it with apktool and add the line you mentioned in the manifest, but it will not re-build, even without adding the line. Do you have a working copy I could get from you?

@Fleker I have a few apps that do not show, like yatse, but yet I can create a lin to them in Kodi or something and they run fine. I really would like to be able to create the shorcuts and hope someone can fix this or something else soon!

**Comment by xstar97** [comment]
@Wanderer189
Download Android studio and GitHub desktop and either download the project from the browser or use GitHub desktop to sync a copy to you pc and launch Android studio to recompile the app with the change

**Comment by trohnjavolta** [comment]
Could anyone provide a working apk here for people like me who don't know what manifest is and how to build and compile apps on their own? Would be awesome!
Right now some apps worked but most for most of them tvapprepo fast closes on me after selecting "create shortcut".

**Comment by oldboy1988** [comment]
Mod with fix https://cloud.mail.ru/public/8tnX/fAPVpV1A6

**Comment by trohnjavolta** [comment]
Thx for the effort, but upon creating a shortcut the app still fast closes
:(

oldboy1988 <notifications@github.com> schrieb am So., 3. März 2019, 22:17:

**Comment by oldboy1988** [comment]
In mi box 3 work fine

**Comment by TimetoDance** [comment]
^ Only works on mi box 3 then. Does jack shit on the shield tv.

**Comment by Wanderer189** [comment]
Yeah, I can't get any results with it on Shield either (folks claim the mi boxes run authentic Android TV OS, but I have yet for a single person tell me if they allow you to use shared network storage like the Shield does, I think it just a better clone than the others)

**Comment by darkj2k** [comment]
> Mod with fix https://cloud.mail.ru/public/8tnX/fAPVpV1A6
Link is not available. Could you please share it again? TIA.

**Comment by mataniss** [comment]
Here is another link to the apk that works with oreo:
https://drive.google.com/file/d/1jE4P5QsCFjD6l-IM8iLeRaK2bohd9pNw/view?usp=sharing

**Comment by ThePapaw** [comment]
> 
> 
> Here is another link to the apk that works with oreo:

This still doesn't work on a Nvidea Shield

**Comment by JesWhite** [comment]
not working. working apk but not working homescreen shortcut function.

**Comment by ThePapaw** [comment]
Yeah, this hasn't worked since android 8 I believe and no one seems or has attempted to fix it correctly...

**Comment by JesWhite** [comment]
> Yeah, this hasn't worked since android 8 I believe and no one seems or has attempted to fix it correctly...

https://play.google.com/store/apps/details?id=dxidev.sideload.channel.launcher i'm trying this app...
worked fine but i cant fix app's image yet.
i must add manually app images. it will be long time for me.

the best app is TVAppRepo :D for us..
